### DISCUSSION TITLE
Proxy troubleshooting ideas

### DISCUSSION THREAD

**Discussion by lanedsmu** [Budibase/budibase#10681] [Help] [answered]
Hi all,

I'm encountering problems sending traffic through our web proxy, and we'd love any suggestions you can send our way.  -Lane

We are running  v2.6.20 and have the GLOBAL_AGENT_HTTP_PROXY and GLOBAL_AGENT_HTTPS_PROXY env variables set.  They /were/ working until I upgraded our environment on 5/8.  It's likely I broke something in that process, but a reversion to OOB containers doesn't appear to have fixed the problem.

We've followed https://docs.budibase.com/docs/running-budibase-behind-company-proxy

we just get timeout errors when trying to execute a REST query.  Our proxy server logs show no requests to/through it. I do know the proxy settings are working for portal functionality:  we can access the templates.

Internal URLs (those matching GLOBAL_AGENT_NO_PROXY) also work quite well.

I also see this in the messages log:
{"level":"ERROR","timestamp":"2023-05-19T15:27:29.641Z","tenantId":"default","msg":"(node:118) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.\n\nPlease migrate your code to use AWS SDK for JavaScript (v3).\nFor more information, check the migration guide at https://a.co/7PzMCcy/n(Use node --trace-warnings ... to show where the warning was created)"}

{"level":"WARN","timestamp":"2023-05-19T15:28:49.101Z","err":{"type":"Error","message":"Query response time exceeded 10000ms timeout.","stack":"Error: Query response time exceeded 10000ms timeout.\n    at /app/dist/threads/index.js:93:32\n    at /app/node_modules/worker-farm/lib/farm.js:199:19\n    at processTicksAndRejections (internal/process/task_queues.js:77:11)","expose":true,"statusCode":400,"status":400},"tenantId":"default","correlationId":"15a2f8d7-3e95-4ea8-a8a0-eacfef61959a","msg":""}
{"level":"INFO","timestamp":"2023-05-19T15:28:49.102Z","req":{"method":"POST","url":"/api/queries/preview","correlationId":"15a2f8d7-3e95-4ea8-a8a0-eacfef61959a"},"res":{"status":400},"responseTime":10084,"msg":"request completed"}
![image](https://github.com/Budibase/budibase/assets/80843079/a111ddbe-623a-4240-b639-850287c7d7ec)

**Comment by lanedsmu** [comment]
More information on this, which I think is making it sound more like a bug, or a very specific misconfiguration:  in addition to the portal functionality being able to access the outbound proxy successfully, app datasources like Google Sheets are able to get out to Google just fine.  It apears it's just the REST datasources that are unable to get out.  

is there additional troubleshooting I can access to get more information about this problem?  I'm assuming others aren't encountering this with the latest builds, else I feel confident there'd be lots of folks reporting this same behavior.

**Reply by lanedsmu** [reply]
Hrm. I just saw https://github.com/Budibase/budibase/issues/7101#issuecomment-1375389185, which makes me wonder if perhaps this is in fact a bug.  

I see this has been tagged as "out of scope", but I'm beginning to wonder if that's the case.  Any update on #7101 ?

**Reply by shogunpurple** [reply]
Hi @lanedsmu - thanks for typing up the detailed discussion. You are correct. We haven't made the changes yet to allow the REST connector to pass through the proxy, as we would need to rewrite our connector to use a different library. We don't have a timeline for this yet, but likely over the next couple of months.

**Reply by lanedsmu** [reply, marked as answer]
ah, ok. That's super helpful information.  I'll subscribe to that issue so we can jump on a release that addresses that when available.
